set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java

```diff
@@ -89,6 +92,26 @@ public PersistentUserSet getByIdentifier(String identifier) {
 		return getUserSetDao().findOne(query);
 	}
 	
+	/* (non-Javadoc)
+	 * @see eu.europeana.set.mongo.service.PersistentUserSetService#isTypeAndCreatorExisting(java.lang.String, eu.europeana.set.definitions.model.agent.Agent)
+	 */
+	public boolean isTypeAndCreatorExisting(String type, Agent creator) {
+	    Query<PersistentUserSet> query = getUserSetDao().createQuery();
+	    query.filter(PersistentUserSet.FIELD_TYPE, type);
+
+	    QueryResults<PersistentUserSet> results = getUserSetDao().find(query);
```

Comment:
the query must return 0 or 1 result

Reply:
query returns long value

Reply:
it must return the object, and not how many objects, the application must ensure that there is only 1 or 0 BookmarksFolders for any user